File: sample-scripts/buy-number/Subscribe_Phone_Num.py
"""
Sample script to purchase a phone number.
Use the /availablephonenumber endpoint to get a list of numbers available for purchase

Read more at https://docs.sonetel.com
"""
import json
import requests
import jwt

# Numumber to purchase
phnum = "ENTER_PHONE_NUMBER"

base_url = "https://public-api.sonetel.com/"

#  API Access token
access_token = "ENTER_ACCESS_TOKEN"
decoded_token = jwt.decode(
    access_token,
    audience='api.sonetel.com',
    options={"verify_signature": False}
    )
acc_id = decoded_token['acc_id']

# Request headers
headers = {
    "Authorization": "Bearer {}".format(access_token),
    'Content-Type': 'application/json'
    }

# Purchase phone number
response = requests.request(
    "POST",
    '{}account/{}/phonenumbersubscription'.format(base_url,acc_id),
    headers=headers,
    data=json.dumps({"phnum": phnum}), timeout=60)

response.raise_for_status()

if response.ok:
    print(response.json())

File: sample-scripts/buy-number/Subscribe_Phone_Num.js
// #  API Access token
var access_token = "ENTER_ACCESS_TOKEN"

// #  Phone numumber to purchase
var phnum = "ENTER_PHONE_NUMBER"

// # Sonetel account ID
var acc_id =  "ENTER_ACCOUNT_ID"

var request = require('request');
var options = {
  'method': 'POST',
  'url': 'https://public-api.sonetel.com/account/'+acc_id+'/phonenumbersubscription',
  'headers': {
    'Authorization': 'Bearer '+ access_token,
    'Content-Type': 'application/json'
  },
  body: JSON.stringify({
    "phnum": phnum
  })

};
request(options, function (error, response) {
  if (error) throw new Error(error);
  console.log(response.body);
});
